src/Microsoft.DotNet.Build.Tasks/PackageFiles/resources.targets

```diff
@@ -9,11 +9,38 @@
 
   <PropertyGroup>
     <ResourcesSourceOutputDirectory Condition="'$(ResourcesSourceOutputDirectory)' == ''">$(MSBuildProjectDirectory)/Resources/</ResourcesSourceOutputDirectory>
-    <StringResourcesPath Condition="'$(StringResourcesPath)'=='' And Exists('$(ResourcesSourceOutputDirectory)Strings.resx')">$(ResourcesSourceOutputDirectory)/Strings.resx</StringResourcesPath>
+    <StringResourcesPath Condition="'$(StringResourcesPath)'=='' And Exists('$(ResourcesSourceOutputDirectory)Strings.resx')">$(ResourcesSourceOutputDirectory)Strings.resx</StringResourcesPath>
     <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.csproj'">$(IntermediateOutputPath)SR.cs</IntermediateResOutputFileFullPath>
     <IntermediateResOutputFileFullPath Condition="'$(MSBuildProjectExtension)' == '.vbproj'">$(IntermediateOutputPath)SR.vb</IntermediateResOutputFileFullPath>
     <DefineConstants Condition="'$(ConfigurationGroup)' == 'Debug'">$(DefineConstants);DEBUGRESOURCES</DefineConstants>
   </PropertyGroup>
+
+  <Target Name="CopyResxFilesToReswFiles" Condition="'$(BuildingUAPVertical)' == 'true'" BeforeTargets="CoreCompile">
+
+    <ItemGroup Condition="'$(BuildingUAPVertical)' == 'true'">
+      <AllResxFiles Include="@(EmbeddedResource)" Condition="'%(Extension)' == '.resx'">
+        <ReswName Condition="'%(EmbeddedResource.ReswName)' == ''">%(EmbeddedResource.ManifestResourceName)</ReswName>
+      </AllResxFiles>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <ProjectHasResources Condition="'@(AllResxFiles)' != ''">true</ProjectHasResources>
+      <TestProjectNeedsModifiedPriFile Condition="'$(IsTestProject)' == 'true' AND '$(ProjectHasResources)' == 'true' AND '$(ResourcesShouldGoToCommonPri)' != 'true'">true</TestProjectNeedsModifiedPriFile>
+      <TestResourcesFolderPath Condition="'$(TestProjectNeedsModifiedPriFile)' == 'true'">$(RuntimePath)$(AssemblyName).Resw</TestResourcesFolderPath>
+      <ResWDestinationPath Condition="'$(TestResourcesFolderPath)' != ''">$(TestResourcesFolderPath)</ResWDestinationPath>
+      <ResWDestinationPath Condition="'$(TestResourcesFolderPath)' == ''">$(ResourcesFolderPath)</ResWDestinationPath>
+    </PropertyGroup>
+
+    <Copy SourceFiles="@(AllResxFiles->'%(FullPath)')"
+          DestinationFiles="@(AllResxFiles->'$(ResWDestinationPath)/%(ReswName).resw')"
+          SkipUnchangedFiles="true">
+          <Output TaskParameter="CopiedFiles" ItemName="FilesCreated" />
+    </Copy>
+
+    <!-- We need to touch the copied files so that the target that uses them can track the inputs and outputs with the copied timestamp -->
+    <Touch Files="@(FilesCreated)" />
```

Comment:
The problem that we where having here is that the Copy task preserves the timestamp of the original copied file so when building only one src assembly we would copy over that resx file if it exists with and old timestamp. If we run the tests then the common pri file will be created with a recent timestamp. However if we after that build another src assembly that has a resx file a new resw file will be created but with a timestamp older than the common pri file that was created previously, so if we build any other tests the common pri will not be updated because the MakeCommonPriFile target's inputs  are older than the output so the task would be skipped. So the only good solution that we found was touching the resw files so that their timestamp is the time when we copied them and if we add a new one after we create the common pri file it will have a newer timestamp than the common pri file and it will be updated.